src/state/chart_state.ts

```diff
@@ -399,10 +399,18 @@ export const chartStoreReducer = (chartId: string) => {
           ...state,
           colors: {
             ...state.colors,
-            persisted: {
-              ...state.colors.persisted,
-              [action.key]: action.color,
-            },
+            persisted:
+              action.color !== null
+                ? {
+                    ...state.colors.persisted,
+                    [action.key]: action.color,
+                  }
+                : (() => {
+                    const colors = { ...state.colors.persisted };
+                    delete colors[action.key];
+
+                    return colors;
```

Comment:
Instead of using delete you can also use the following:
```suggestion
                    const {[action.key] : removed, ...others} = state.colors.persisted;
                    return others;
```

Reply:
8f9a229